src/controllers/restful/PaymentController.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * PaymentControllers
+ * RESTful Controllers for Payments Endpoints
+ */
+
+import * as express from "express";
+import { Either, left, right } from "fp-ts/lib/Either";
+import { PPTPortTypes } from "italia-pagopa-api/dist/wsdl-lib/PagamentiTelematiciPspNodoservice/PPTPort";
+import * as uuid from "uuid";
+import { PagoPaConfig } from "../../Configuration";
+import { ControllerError } from "../../enums/ControllerError";
+import { HttpErrorStatusCode } from "../../enums/HttpErrorStatusCode";
+import * as PaymentsService from "../../services/PaymentsService";
+import { PaymentsActivationRequest } from "../../types/controllers/PaymentsActivationRequest";
+import { PaymentsActivationResponse } from "../../types/controllers/PaymentsActivationResponse";
+import { PaymentsCheckRequest } from "../../types/controllers/PaymentsCheckRequest";
+import { PaymentsCheckResponse } from "../../types/controllers/PaymentsCheckResponse";
+import { CodiceContestoPagamento } from "../../types/PagoPaTypes";
+import * as PaymentsConverter from "../../utils/PaymentsConverter";
+import * as RestfulUtils from "../../utils/RestfulUtils";
+// Forward a payment check request from BackendApp to PagoPa
+export async function checkPaymentToPagoPa(
```

Comment:
we need MUCH MUCH more comments in the body of this function, you must describe exactly of the steps map to the pagoPA APIs - personally even I have have familiarity with pagoPA, I can't mentally map what is going on here to the pagoPA process (also because the names of the API methods are different)